fathom/src/core/binary.rs

```diff
@@ -375,3 +385,23 @@ read_multibyte_prim!(read_f32le, from_le_bytes, f32);
 read_multibyte_prim!(read_f32be, from_be_bytes, f32);
 read_multibyte_prim!(read_f64le, from_le_bytes, f64);
 read_multibyte_prim!(read_f64be, from_be_bytes, f64);
+
+impl fmt::Display for ReadError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ReadError::Io(err) => err.fmt(f),
+            ReadError::Fail => f.write_str("format failed"),
+            ReadError::CondFail => f.write_str("conditional format failed"),
+            ReadError::UnexpectedError(msg) => f.write_str(msg),
+            ReadError::InvalidFormat => f.write_str("invalid format"),
+        }
+    }
+}
+
+impl std::error::Error for ReadError {}
+
+impl From<io::Error> for ReadError {
+    fn from(err: io::Error) -> ReadError {
+        ReadError::Io(err)
+    }
+}
```

Comment:
Wondering if these impls should be grouped with `ReadError`. 🤔

Reply:
As in move them up to the struct or move the struct down to the impls?

Reply:
Maybe grouped with the enum just to keep things in one place.